src/journal/journal-file.c

```diff
@@ -3970,6 +4075,26 @@ bool journal_file_rotate_suggested(JournalFile *f, usec_t max_file_usec) {
                         return true;
                 }
 
+        /* If there are too many hash collisions somebody is most likely playing games with us. Hence, if our
+         * longest chain is longer than 2/3rds of the hash table size, let's suggest rotation. */
```

Comment:
If nobody is "playing games" i.e. the hash values are uniformly distributed, then the longest chain should scale as the logarithm of the hash table size.

Reply:
Rather than using the logarithm, you could also use a hard-coded value that is unlikely to be reached even with the largest possible hash table size.

e.g. for a fill factor of 75% and a hash table size of 1 billion, the chance of a hash chain of length 100 is approximately:

    1e9 * pow(0.75, 100)
    == 3e-4

Since the consequences of a false positive here are not severe (it will just rotate the logs and start a new journal file), that might be the simplest solution.

Reply:
Well, but given that our hash tables are picked in a constant way, the threshold here already *is* constant...

Reply:
True, but it doesn't change the fact that a threshold of N*2/3 is too large to defend against the O(N) worst case insertion behaviour.

Using the actual hash table size 233016, rather than the 1 billion I assumed earlier, we can set the threshold at 75 items and have a false positive rate less than 0.01%:
```
233016 * pow(0.75, 75)
== 9.9e-05
```

Reply:
or keeping 100 items, the false positive rate will be less than 1 in 10 million

Reply:
ok, will change the threshold to a fixed 100...

But i mean, we are talking about some constant here... any constant would really work here, given we just want to keep behaviour in O(1) instead of O(n)